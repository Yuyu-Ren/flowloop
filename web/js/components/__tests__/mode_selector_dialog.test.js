/* eslint-env jest */

import ModeSelectorDialog from '../mode_selector_dialog'

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

describe('ModeSelectorDialog', () => {
  it('exists', () => {
    expect(ModeSelectorDialog).toBeTruthy()
  })

  it('renders', () => {
    const comp = <ModeSelectorDialog />
    const wrapper = shallow(comp)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders with CONTINUOUS', () => {
    const comp = <ModeSelectorDialog selected='CONTINUOUS' />
    const wrapper = shallow(comp)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('is clickable', () => {
    return new Promise((resolve, reject) => {
      const comp = <ModeSelectorDialog selected='CONTINUOUS' onSelect={onSelect} />
      const wrapper = shallow(comp)
      wrapper.find('.option.-single').simulate('click')

      function onSelect () {
        expect(shallowToJson(wrapper)).toMatchSnapshot()
        resolve()
      }
    })
  })
})
