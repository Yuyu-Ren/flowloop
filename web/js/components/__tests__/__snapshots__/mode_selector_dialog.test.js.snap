// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeSelectorDialog is clickable 1`] = `
<div
  className="mode-selector-dialog"
>
  <h1
    className="heading"
  >
    Select a timer mode
  </h1>
  <div
    className="options"
  >
    <button
      className="option mode-selector-option -continuous -active"
      onClick={[Function]}
    >
      <span
        className="icon -continuous"
      />
      <h2>
        Continuous
      </h2>
      <h5>
        Flow mode
      </h5>
      <p>
        Timers never end. Laps are recorded every time a timer completes.
      </p>
    </button>
    <button
      className="option mode-selector-option -single"
      onClick={[Function]}
    >
      <span
        className="icon -single"
      />
      <h2>
        Single
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Stops the timer after it completes. Works like classic Pomodoro.
      </p>
    </button>
    <button
      aria-label="Coming soon. Stay tuned!"
      className="option mode-selector-option -alternate hint--bottom"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="icon -alternate"
      />
      <h2>
        Alternating
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Switches to a break after a work timer completes, and vice versa.
      </p>
    </button>
  </div>
</div>
`;

exports[`ModeSelectorDialog renders 1`] = `
<div
  className="mode-selector-dialog"
>
  <h1
    className="heading"
  >
    Select a timer mode
  </h1>
  <div
    className="options"
  >
    <button
      className="option mode-selector-option -continuous"
      onClick={[Function]}
    >
      <span
        className="icon -continuous"
      />
      <h2>
        Continuous
      </h2>
      <h5>
        Flow mode
      </h5>
      <p>
        Timers never end. Laps are recorded every time a timer completes.
      </p>
    </button>
    <button
      className="option mode-selector-option -single"
      onClick={[Function]}
    >
      <span
        className="icon -single"
      />
      <h2>
        Single
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Stops the timer after it completes. Works like classic Pomodoro.
      </p>
    </button>
    <button
      aria-label="Coming soon. Stay tuned!"
      className="option mode-selector-option -alternate hint--bottom"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="icon -alternate"
      />
      <h2>
        Alternating
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Switches to a break after a work timer completes, and vice versa.
      </p>
    </button>
  </div>
</div>
`;

exports[`ModeSelectorDialog renders with CONTINUOUS 1`] = `
<div
  className="mode-selector-dialog"
>
  <h1
    className="heading"
  >
    Select a timer mode
  </h1>
  <div
    className="options"
  >
    <button
      className="option mode-selector-option -continuous -active"
      onClick={[Function]}
    >
      <span
        className="icon -continuous"
      />
      <h2>
        Continuous
      </h2>
      <h5>
        Flow mode
      </h5>
      <p>
        Timers never end. Laps are recorded every time a timer completes.
      </p>
    </button>
    <button
      className="option mode-selector-option -single"
      onClick={[Function]}
    >
      <span
        className="icon -single"
      />
      <h2>
        Single
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Stops the timer after it completes. Works like classic Pomodoro.
      </p>
    </button>
    <button
      aria-label="Coming soon. Stay tuned!"
      className="option mode-selector-option -alternate hint--bottom"
      disabled={true}
      onClick={[Function]}
    >
      <span
        className="icon -alternate"
      />
      <h2>
        Alternating
      </h2>
      <h5>
        Pomodoro mode
      </h5>
      <p>
        Switches to a break after a work timer completes, and vice versa.
      </p>
    </button>
  </div>
</div>
`;
